examples/dump_media/main.py

```diff
@@ -0,0 +1,201 @@
+                        media["id"]
+                        for media in response.json()["media"]
+                        if MediaDownloader.is_media_downloadable(media)
+                    ]
+                )
+        logger.info(f"Found media ids: {media_ids}")
+        return media_ids
+
+    def search_media(self, page):
+        self.logger.info(f"Getting {page}. page")
+        response = self.public_api_client.request(
+            "get", "media/search", params={"page": page}
+        )
+        self.raise_exception_if_not_ok(
+            response, f"Could not get all media on page {page}: {response.text}"
+        )
+        return response
+
+    def raise_exception_if_not_ok(self, response, error_message):
+        if response.status_code != 200:
+            self.logger.error(f"Abort: {response}")
+            raise Exception(error_message)
+
+    @staticmethod
+    def is_media_downloadable(media):
+        return media["transcoding_status"] == "transcoding_finished"
+
+    @log_exception
+    def extend_all_media_with_collection_and_captions(self, media_ids):
+        media_objects = []
+        for id in media_ids:
+            media_with_details = self.get_details_of_media(id)
+            captions = self.get_captions_of_media(id)
+            media_with_details.update(captions)
+            media_objects.append(media_with_details)
+        return media_objects
+
+    def get_details_of_media(self, media_id):
+        self.logger.info(f"Getting details of media: {media_id}")
+        response = self.public_api_client.request("get", f"media/{media_id}")
+        self.raise_exception_if_not_ok(
+            response, f"Could not get details for media with id {id}: {response.text}"
+        )
+        return response.json()["media"]
+
+    def get_captions_of_media(self, media_id):
+        self.logger.info(f"Getting caption data for media: {media_id}")
+        response = self.public_api_client.request(
+            "get", f"media/{media_id}/caption_files"
+        )
+        self.raise_exception_if_not_ok(
+            response,
+            f"Could not get caption files for media with id {id}: {response.text}",
+        )
+
+        captions = [
+            caption
+            for caption in response.json()["caption_files"]
+            if (caption["status"] == "published" or caption["status"] == "uploaded")
+        ]
+
+        return {"caption_files": captions}
+
+    @log_exception
+    def download_all_media(self, media_objects):
+        for media in media_objects:
+            user_name = media["owner"]["full_name"]
+            collection_name = media["collection"]["name"]
+
+            collection_path = self.root_dir / user_name / collection_name
+            collection_path.mkdir(parents=True, exist_ok=True)
+
+            if not self.dry_run:
+                self.download_a_media(collection_path, media)
+            else:
+                print("--dry_run was on: creating directory structure only")
```

Comment:
This line will print the "--dry_run was on: creating directory structure only" line for each media. imho it needs to be moved out of the for loop as

```
            if self.dry_run:
                print("--dry_run was on: creating directory structure only")
```

Reply:
true!